app/assets/javascripts/initializers/initializeSwipeGestures.js

```diff
@@ -1,61 +1,116 @@
-function initializeSwipeGestures(){
-  initializeSwipeGestures.called = true;
-  swipeState = "middle";
-  setTimeout(function(){
+/* global slideSidebar */
+'use strict';
 
+function initializeSwipeGestures() {
+  initializeSwipeGestures.called = true;
+  let swipeState = 'middle';
+  setTimeout(function () {
     // window.onload=function(){
-    (function(d){
-    var
-    ce=function(e,n){var a=document.createEvent("CustomEvent");a.initCustomEvent(n,true,true,e.target);e.target.dispatchEvent(a);a=null;return false},
-    nm=true,sp={x:0,y:0},ep={x:0,y:0},
-    touch={
-      touchstart:function(e){sp={x:e.touches[0].pageX,y:e.touches[0].pageY,scrollY:window.scrollY}},
-      touchmove:function(e){nm=false;ep={x:e.touches[0].pageX,y:e.touches[0].pageY,scrollY:window.scrollY}},
-      touchend:function(e){
-        if(nm){ce(e,'fc')}
-        else {
-          var x=ep.x-sp.x,xr=Math.abs(x),y=ep.y-sp.y,yr=Math.abs(y),absScroll=Math.abs(sp.scrollY-ep.scrollY);
-          if(Math.max(xr,yr)>15){
-            var shouldScroll = (xr/2)>yr && absScroll < 5;
-            ce(e,(shouldScroll?(x<0?'swl':'swr'):(y<0?'swu':'swd')))
-          }
+    (function (d) {
```

Comment:
do you think we could move this function in a named function and then call it on line 68? It should probably fix Codeclimate error as well.

cc @nickytonline

Reply:
I feel that the `function (d)` as a whole forms an integral part the `initializeSwipeGestures()`... Rather moving the function as a whole, moving the eventHandlers for `touchstart`, `touchend`, `touchmove` and `touchcancel` should make more sense as independent units.

Also, the independent `function (d)` will exceed the codeclimate 50 line limit as well.

Reply:
@rhymes, this code looks like it was pulled from somewhere that had it minified already. Since this PR is just about fixing the lint issues, I would address this in another PR if we are to pursue it.